src/Mafia/Cabal/Sandbox.hs

```diff
@@ -62,17 +61,42 @@ dropLeft :: Either a b -> Maybe b
 dropLeft =
   either (const Nothing) Just
 
+removeSandbox :: EitherT CabalError IO ()
+removeSandbox = do
+  -- remove sandbox directory and sandbox config file
+  dir <- getInferredSandboxDir
+  ignoreIO $ removeDirectoryRecursive dir
+  ignoreIO $ removeFile defaultConfig
+
+  -- remove root sandbox directory if it is now empty
+  whenM (doesDirectoryExist sandboxRoot) . ignoreIO $ do
```

Comment:
I might just make this whole function assume a single thread, the rest of cabal probably can't handle multiple builds at the same time anyway